spec/controllers/auth_key_pair_cloud_controller_spec.rb

```diff
@@ -1,61 +1,52 @@
 describe AuthKeyPairCloudController do
-  context "#button" do
-    before(:each) do
-      stub_user(:features => :all)
-      EvmSpecHelper.create_guid_miq_server_zone
-    end
+  let(:kp) { FactoryGirl.create(:auth_key_pair_cloud, :name => "auth-key-pair-cloud-01") }
+  let(:classification) { FactoryGirl.create(:classification, :name => "department", :description => "Department") }
+  let(:tag1) { FactoryGirl.create(:classification_tag, :name   => "tag1", :parent => classification) }
+  let(:tag2) { FactoryGirl.create(:classification_tag, :name   => "tag2", :parent => classification) }
+  let!(:user) { stub_user(:features => :all) }
+
+  before do
+    EvmSpecHelper.create_guid_miq_server_zone
+    FactoryGirl.create(:tagging, :tag => tag1.tag, :taggable => kp)
+    FactoryGirl.create(:tagging, :tag => tag2.tag, :taggable => kp)
   end
 
-  context "#tags_edit" do
-    let!(:user) { stub_user(:features => :all) }
-    before(:each) do
-      EvmSpecHelper.create_guid_miq_server_zone
-      @kp = FactoryGirl.create(:auth_key_pair_cloud, :name => "auth-key-pair-cloud-01")
-      allow(@kp).to receive(:tagged_with).with(:cat => user.userid).and_return("my tags")
-      classification = FactoryGirl.create(:classification, :name => "department", :description => "Department")
-      @tag1 = FactoryGirl.create(:classification_tag,
-                                 :name   => "tag1",
-                                 :parent => classification)
-      @tag2 = FactoryGirl.create(:classification_tag,
-                                 :name   => "tag2",
-                                 :parent => classification)
-      allow(Classification).to receive(:find_assigned_entries).with(@kp).and_return([@tag1, @tag2])
+  describe "#tags_edit" do
+    before do
       session[:tag_db] = "ManageIQ::Providers::CloudManager::AuthKeyPair"
-      edit = {
-        :key        => "ManageIQ::Providers::CloudManager::AuthKeyPair_edit_tags__#{@kp.id}",
+      session[:edit] = {
+        :key        => "ManageIQ::Providers::CloudManager::AuthKeyPair_edit_tags__#{kp.id}",
         :tagging    => "ManageIQ::Providers::CloudManager::AuthKeyPair",
-        :object_ids => [@kp.id],
+        :object_ids => [kp.id],
         :current    => {:assignments => []},
-        :new        => {:assignments => [@tag1.id, @tag2.id]}
+        :new        => {:assignments => [tag1.id, tag2.id]}
       }
-      session[:edit] = edit
-    end
-
-    after(:each) do
-      expect(response.status).to eq(200)
     end
 
     it "builds tagging screen" do
-      post :button, :params => { :pressed => "auth_key_pair_cloud_tag", :format => :js, :id => @kp.id }
+      post :button, :params => { :pressed => "auth_key_pair_cloud_tag", :format => :js, :id => kp.id }
```

Comment:
Just a minor styling thing and I'll leave it up to you, but having a blank line after the `get/post` looks a little bit better.
IMO most readable way is like this:
```ruby
it 'something' do
  # setting up context that couldn't go to a before/let block

  # firing the request

  # expectations
end
```